plugins/inputs/procstat/procstat.go

```diff
@@ -367,16 +388,34 @@ func (p *Procstat) getPIDFinder() (PIDFinder, error) {
 }
 
 // Get matching PIDs and their initial tags
-func (p *Procstat) findPids() ([]PID, map[string]string, error) {
-	var pids []PID
-	tags := make(map[string]string)
-	var err error
+func (p *Procstat) findPids() []PidsTags {
+	var pidTags []PidsTags
 
 	f, err := p.getPIDFinder()
 	if err != nil {
-		return nil, nil, err
+		pidTags = append(pidTags, PidsTags{nil, nil, err})
```

Comment:
I assume it's intentional that we're continuing after the error?

Reply:
Good catch, it wasn't intentional. I moved the `p.getPIDFinder` call closer to where the returned value is being used and return if an error is found.